src/inspector_agent.cc

```diff
@@ -706,6 +773,64 @@ void Url(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), url.c_str()));
 }
 
+static void InvokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
+                                    const FunctionCallbackInfo<Value>& args);
+
+static void* GetAsyncTask(intptr_t asyncId);
+
+static void AsyncTaskScheduledWrapper(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  CHECK(args[1]->IsNumber());
+  intptr_t task_id = args[1]->IntegerValue(env->context()).FromJust();
```

Comment:
This goes back to your original comment in https://github.com/nodejs/node/pull/13870#discussion_r124930411.

I acknowledge the truncation can be a problem in theory. In practice, we should be fine because of  https://github.com/nodejs/node/pull/13870#discussion_r126054161. Even if it was not true, I personally think it's better to have async stack traces in the first ~20 minutes of the debugging session than to not have any async stack traces at all.

However, to make this truncation issue more obvious, I'll use `int64_t` type here, and move the truncation into `GetAsyncTask` function, together with an explanatory comment.